docs/diga_api.md

```diff
@@ -0,0 +1,289 @@
+                         version="002.000.000" gueltigab="2020-07-01" verfahrenskennung="TDFC0" nachrichtentyp="ANT"
+                         absender="987654321" empfaenger="123456789">
+    <Antwort>
+        <IK_DiGA_Hersteller>123456789</IK_DiGA_Hersteller>
+        <IK_Krankenkasse>987654321</IK_Krankenkasse>
+        <DiGAVEID>12345000</DiGAVEID>
+        <Freischaltcode>ABCDEFGHIJKLMNOP</Freischaltcode>
+        <Tag_der_Leistungserbringung>2020-08-19</Tag_der_Leistungserbringung>
+    </Antwort>
+</Pruefung_Freischaltcode>
+```
+
+The response differs from the request by having `nachrichtentyp="ANT"` as header (`Antwort`) instead of `nachrichtentyp="ANF"` (`Anfrage`).
+Also the diga id is listed as `DiGAVEID` (verified ID) and an additional field `Tag_der_Leistungserbringung` is returned (day when the service started).
+
+## Billing requests
+
+Invoices must be created using the `XRechnung` standard.
+There might also be dedicated accounting/book-keeping software which allows creating and sending invoices according to this standard, so that billing might also be handled outside the production system.
+However, the client supports sending invoices by mail.
+It is up to the users of the client to define what should happen with the response from the diga api.
+E.g. invoices are likely to be required for accounting and therefore must be stored.
+Also handling of apis which do not support billing yet, must be defined.
+
+The invoice should contain the following information:
+
+- prescription code
+- DiGA id of the manufacturer
+- time period for prescription
+- information about the invoice issuer and the insurer
+
+Additional information can be found on this [wiki page](https://github.com/alex-therapeutics/diga-api-client/wiki/Billing-validation---XRechnung).
+
+## Encryption
+
+Given that we are dealing with patient data, it is required to properly encrypt the data sent within requests.
+The encryption is based on certificates which need to be requested at the ITSG which acts as a trust center.
+You can request it at [ITSG website](https://www.itsg.de/produkte/trust-center/zertifikat-beantragen/), assuming that you already have an IK number.
+
+A technical documentation for the encryption (of course in German) can be found on the [technical standards page at gkv](https://www.gkv-datenaustausch.de/technische_standards_1/technische_standards.jsp) as [attachment 16](https://www.gkv-datenaustausch.de/media/dokumente/standards_und_normen/technische_spezifikationen/Anlage_16.pdf) under `Security Schnittstelle (SECON)` section.
+Once the certificates are issued, checkout the [prerequisites from the Readme](https://github.com/alex-therapeutics/diga-api-client#prerequisites) to be able to use them with the api client.
+
+## Request processing time
+
+According to the documentation, the processing time constraints are as follows:
+
+Mean value 10000[msec], 95% quantile [msec] 15000
+
+Is that really 10 seconds for a single request?
```

Comment:
Is it really that bad?

Reply:
We found this information [here](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_Anlage_1_Technische_Anlage_zur_RL_V1.1_20210225.pdf) in section 5.6.5.

Based on all the other documentation we understood that the mean response is 10s which was a bit of shocker. Glad to here it's not the case :)

Reply:
Removed this section now